### PR TITLE
Add ItemMetaMock.getLocalizedName and ItemMetaMock.setLocalizedName

### LINKED ISSUE #265: Add ItemMetaMock.getLocalizedName and ItemMetaMock.setLocalizedName
Similar to `ItemMetaMock.displayName`.

### PR DESCRIPTION
**Pull request by sya-ri**
# Description

Closes #265  

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Version number updated in `settings.gradle`.
- [x] Unit tests added.
- [x] Code follows existing code format.

# Info on creating a pull request
Before the pull request can be accepted, the version number stored in settings.gradle should be updated.
MockBukkit uses semantic-versioning (https://semver.org/)

 - If the PR fixes a bug in MockBukkit, update the patch version. (if the current version is `0.4.1`, the new version should be `0.4.2`)
 - If the PR adds a new feature to MockBukkit, update minor version. (if the current version is `0.4.1`, the new version should be `0.5.0`)
 - Make sure that unit tests are added which test the relevant changes.
 - Make sure that the changes follow the existing code format.

Note that a PR that fixes an `UnimplementedOperationException` should be considered a new version and not a bugfix.

The version number can be found in `settings.gradle`.

### CODE REVIEW

File: src/test/java/be/seeseemelk/mockbukkit/inventory/meta/ItemMetaMockTest.java [hunk 1]
```diff
@@ -392,6 +392,14 @@ void getLore_LoreChangedAfterSet_LoreNotChanged()
 		assertEquals("world", lore.get(1));
 	}
 
+	@Test
+	void setLocalizedName_NewName_NameSetExactly()
+	{
+		meta.setLocalizedName("Some name");
+		assertTrue(meta.hasLocalizedName());
+		assertEquals("Some name", meta.getLocalizedName());
+	}
```
Comment (TheBusyBiscuit) [inline src/test/java/be/seeseemelk/mockbukkit/inventory/meta/ItemMetaMockTest.java:407]: An additional unit test for the default behaviour would be nice.
In other words: assertFalse for `meta.hasLocalizedName()` and asserting that the localized name is null by default.

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]
Looks pretty much good to me, simple and sleek.
An additional unit test would be nice to have though!

File: src/test/java/be/seeseemelk/mockbukkit/inventory/meta/ItemMetaMockTest.java (hunk 1 above)
Comment (sya-ri) [reply src/test/java/be/seeseemelk/mockbukkit/inventory/meta/ItemMetaMockTest.java:407]: fix [`398c354` (#268)](https://github.com/MockBukkit/MockBukkit/pull/268/commits/398c354401cee6db7337304bfaefc0076729c8a1)

File: src/main/java/be/seeseemelk/mockbukkit/inventory/meta/ItemMetaMock.java [hunk 2]
```diff
@@ -427,22 +426,19 @@ public static ItemMetaMock deserialize(Map<String, Object> args)
...
 	@Override
 	public String getLocalizedName()
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		return localizedName;
 	}
 
 	@Override
 	public void setLocalizedName(String name)
```
Comment (TheBusyBiscuit) [inline src/main/java/be/seeseemelk/mockbukkit/inventory/meta/ItemMetaMock.java]:
```suggestion
	public void setLocalizedName(@Nullable String name)
```

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]
I looked at the Spigot docs and found out that this is indeed nullable.
An additional test for setting the localized name to null and confirming that `hasLocalizedName` is false would be appropriate to cover all cases.

This should be the last thing though, after that it looks to be ready for merging.

**Review by seeseemelk — APPROVED** [review]

**Review by TheBusyBiscuit — APPROVED** [review]
LGTM

**Pull request merged**